source/CMakeLists.txt

```diff
@@ -24,6 +24,54 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+set(GRAMMAR_JSON_FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+)
+set(GRAMMAR_INC_FILES
+  ${spirv-tools_BINARY_DIR}/core.insts.inc
+  ${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+  ${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+)
+set(GRAMMAR_PROCESSING_SCRIPT
+  ${spirv-tools_SOURCE_DIR}/utils/generate_grammar_tables.py)
+
+add_custom_command(OUTPUT ${GRAMMAR_INC_FILES}
+  COMMAND ${PYTHON_EXECUTABLE} ${GRAMMAR_PROCESSING_SCRIPT}
+    --spirv-core-grammar=${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+    --extinst-glsl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+    --extinst-opencl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+    --core-insts-output=${spirv-tools_BINARY_DIR}/core.insts.inc
+    --glsl-insts-output=${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+    --opencl-insts-output=${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+    --operand-kinds-output=${spirv-tools_BINARY_DIR}/operand.kinds.inc
+  DEPENDS ${GRAMMAR_PROCESSING_SCRIPT} ${GRAMMAR_JSON_FILES}
+  COMMENT "Generate info tables for SPIR-V instructions and operands." VERBATIM)
+
+# The following .cpp files include the above generated .inc files.
+# Add those .inc files as their dependencies.
```

Comment:
I'm not crystally clear about how CMake manages header dependencies. So please argue if the following doesn't make sense. And I'd like to poke @AWoloszyn for expertise. :)

From my understanding, you almost specify everything. You'll need to list source files for targets, and you'll need to specify how these targets involve with each other. But it do works if you omit some header files as dependencies for targets; they are not explicitly listed but get added in those `.d` files: I think that's because CMake leverages the power of GCC/Clang. (?)

For the `.inc` files here, they are generated by a custom Python script. Although they are included similarly as typical `.h` files, but apparently the compiler knows nothing about this extension and I doubt the compiler will give any sort of help here. So, what information we end up give CMake about these `.inc` files? CMake only sees that a custom command generates some `.inc` files that get no users. So it will not generate these `.inc` files by itself. And later when the compiler searches the `-I` directories, which is given by CMake, none of these `.inc` files will be found.